### PR TITLE
Use numpy & dask sliding_window_view for rolling

### LINKED ISSUE #2514: interpolate_na with limit argument changes size of chunks
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd
import xarray as xr
import numpy as np

t = pd.date_range(start='2018-01-01', end='2018-02-01', freq='H')
foo = np.sin(np.arange(len(t)))
bar = np.cos(np.arange(len(t)))

foo[1] = np.NaN
bar[2] = np.NaN

ds_test = xr.Dataset(data_vars={'foo': ('time', foo),
                           'bar': ('time', bar)},
                    coords={'time': t}).chunk()

print(ds_test)
print("\n\n### After `.interpolate_na(dim='time')`\n")
print(ds_test.interpolate_na(dim='time'))
print("\n\n### After `.interpolate_na(dim='time', limit=5)`\n")
print(ds_test.interpolate_na(dim='time', limit=5))
print("\n\n### After `.interpolate_na(dim='time', limit=20)`\n")
print(ds_test.interpolate_na(dim='time', limit=20))
```

Output of the above code. Note the different chunk sizes, depending on the value of `limit`:
```
<xarray.Dataset>
Dimensions:  (time: 745)
Coordinates:
  * time     (time) datetime64[ns] 2018-01-01 2018-01-01T01:00:00 ... 2018-02-01
Data variables:
    foo      (time) float64 dask.array<shape=(745,), chunksize=(745,)>
    bar      (time) float64 dask.array<shape=(745,), chunksize=(745,)>


### After `.interpolate_na(dim='time')`

<xarray.Dataset>
Dimensions:  (time: 745)
Coordinates:
  * time     (time) datetime64[ns] 2018-01-01 2018-01-01T01:00:00 ... 2018-02-01
Data variables:
    foo      (time) float64 dask.array<shape=(745,), chunksize=(745,)>
    bar      (time) float64 dask.array<shape=(745,), chunksize=(745,)>


### After `.interpolate_na(dim='time', limit=5)`

<xarray.Dataset>
Dimensions:  (time: 745)
Coordinates:
  * time     (time) datetime64[ns] 2018-01-01 2018-01-01T01:00:00 ... 2018-02-01
Data variables:
    foo      (time) float64 dask.array<shape=(745,), chunksize=(3,)>
    bar      (time) float64 dask.array<shape=(745,), chunksize=(3,)>


### After `.interpolate_na(dim='time', limit=20)`

<xarray.Dataset>
Dimensions:  (time: 745)
Coordinates:
  * time     (time) datetime64[ns] 2018-01-01 2018-01-01T01:00:00 ... 2018-02-01
Data variables:
    foo      (time) float64 dask.array<shape=(745,), chunksize=(10,)>
    bar      (time) float64 dask.array<shape=(745,), chunksize=(10,)>
```

#### Problem description

When using `xarray.DataArray.interpolate_na()` with the `limit` kwarg this changes the chunksize of the resulting `dask.arrays`.

#### Expected Output

The chunksize should not change. Very small chunks which results from typical small values of `limit` are not optimal for the performance of `dask`. Also, things like `.rolling()` will fail if the chunksize is smaller than the window length of the rolling window.

#### Output of ``xr.show_versions()``

<details>
INSTALLED VERSIONS
------------------
commit: None
python: 2.7.15.final.0
python-bits: 64
OS: Darwin
OS-release: 16.7.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: None
LANG: de_DE.UTF-8
LOCALE: None.None

xarray: 0.10.9
pandas: 0.23.3
numpy: 1.13.3
scipy: 1.0.0
netCDF4: 1.4.1
h5netcdf: 0.5.0
h5py: 2.8.0
Nio: None
zarr: None
cftime: 1.0.1
PseudonetCDF: None
rasterio: None
iris: None
bottleneck: 1.2.1
cyordereddict: 1.0.0
dask: 0.19.4
distributed: 1.23.3
matplotlib: 2.2.2
cartopy: 0.16.0
seaborn: 0.8.1
setuptools: 38.5.2
pip: 9.0.1
conda: 4.5.11
pytest: 3.4.2
IPython: 5.5.0
sphinx: None
</details>

### LINKED ISSUE #2531: DataArray.rolling() does not preserve chunksizes in some cases
This issue was found and discussed in the related issue #2514 

I open a separate issue for clarity.

#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd
import numpy as np
import xarray as xr

t = pd.date_range(start='2018-01-01', end='2018-02-01', freq='H')
bar = np.sin(np.arange(len(t)))
baz = np.cos(np.arange(len(t)))

da_test = xr.DataArray(data=np.stack([bar, baz]),
                       coords={'time': t,
                               'sensor': ['one', 'two']},
                       dims=('sensor', 'time'))

print(da_test.chunk({'time': 100}).rolling(time=60).mean().chunks)

print(da_test.chunk({'time': 100}).rolling(time=60).count().chunks)
```
```
Output for `mean`: ((2,), (745,))
Output for `count`: ((2,), (100, 100, 100, 100, 100, 100, 100, 45))
Desired Output: ((2,), (100, 100, 100, 100, 100, 100, 100, 45))
```
#### Problem description

DataArray.rolling() does not preserve the chunksizes, apparently depending on the applied method.

#### Output of ``xr.show_versions()``

<details>
INSTALLED VERSIONS
------------------
commit: None
python: 2.7.15.final.0
python-bits: 64
OS: Darwin
OS-release: 16.7.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: None
LANG: de_DE.UTF-8
LOCALE: None.None

xarray: 0.10.9
pandas: 0.23.3
numpy: 1.13.3
scipy: 1.0.0
netCDF4: 1.4.1
h5netcdf: 0.5.0
h5py: 2.8.0
Nio: None
zarr: None
cftime: 1.0.1
PseudonetCDF: None
rasterio: None
iris: None
bottleneck: 1.2.1
cyordereddict: 1.0.0
dask: 0.19.4
distributed: 1.23.3
matplotlib: 2.2.2
cartopy: 0.16.0
seaborn: 0.8.1
setuptools: 38.5.2
pip: 9.0.1
conda: 4.5.11
pytest: 3.4.2
IPython: 5.5.0
sphinx: None
</details>

### LINKED ISSUE #3165: rolling: bottleneck still not working properly with dask arrays
#### MCVE Code Sample

```python
# Your code here
import numpy as np
import xarray as xr
# from dask.distributed import Client
temp= xr.DataArray(np.zeros((5000, 50000)),dims=("x","y")).chunk({"y":100, })
temp.rolling(x=100).mean()
```

#### Problem Description

I was thrilled to find that the new release (both 0.12.2 and 0.12.3) fixed the rolling window issue. However, When I tried, it seems the problem is still there. Previously, the above code runs with bottleneck installed. However, with the new version, with or without bottleneck, it simply gives the memory error as below.

I have tried to use old and new versions of Dask and pandas, but with no much difference. However, the dask Dataframe version of the code (shown below) runs ok.
```python
import dask.dataframe as dd
import dask.array as da
import numpy as np

da_array=da.from_array(np.zeros((5000, 50000)), chunks=(5000,100))
df = dd.from_dask_array(da_array)
df.rolling(window=100,axis=0).mean()
```
I have also tried to apply the similar thing on dataset from netcdf files, it simply started consuming very large portion of memory and gives the similar errors.

Any help are appreciated.

```
/miniconda3/envs/xarray/lib/python3.7/site-packages/xarray/core/merge.py:17: FutureWarning: The Panel class is removed from pandas. Accessing it from the top-level namespace will also be removed in the next version
  PANDAS_TYPES = (pd.Series, pd.DataFrame, pd.Panel)
/miniconda3/envs/xarray/lib/python3.7/site-packages/xarray/core/dataarray.py:219: FutureWarning: The Panel class is removed from pandas. Accessing it from the top-level namespace will also be removed in the next version
  elif isinstance(data, pd.Panel):
Traceback (most recent call last):
  File "rolltest.py", line 5, in <module>
    temp.rolling(x=100).mean()
  File "/miniconda3/envs/xarray/lib/python3.7/site-packages/xarray/core/rolling.py", line 245, in wrapped_func
    return self.reduce(func, **kwargs)
  File "/miniconda3/envs/xarray/lib/python3.7/site-packages/xarray/core/rolling.py", line 217, in reduce
    result = windows.reduce(func, dim=rolling_dim, **kwargs)
  File "/miniconda3/envs/xarray/lib/python3.7/site-packages/xarray/core/dataarray.py", line 1636, in reduce
    var = self.variable.reduce(func, dim, axis, keep_attrs, **kwargs)
  File "/miniconda3/envs/xarray/lib/python3.7/site-packages/xarray/core/variable.py", line 1369, in reduce
    input_data = self.data if allow_lazy else self.values
  File "/miniconda3/envs/xarray/lib/python3.7/site-packages/xarray/core/variable.py", line 392, in values
    return _as_array_or_item(self._data)
  File "/miniconda3/envs/xarray/lib/python3.7/site-packages/xarray/core/variable.py", line 213, in _as_array_or_item
    data = np.asarray(data)
  File "/miniconda3/envs/xarray/lib/python3.7/site-packages/numpy/core/numeric.py", line 538, in asarray
    return array(a, dtype, copy=False, order=order)
  File "/miniconda3/envs/xarray/lib/python3.7/site-packages/dask/array/core.py", line 1047, in __array__
    x = self.compute()
  File "/miniconda3/envs/xarray/lib/python3.7/site-packages/dask/base.py", line 156, in compute
    (result,) = compute(self, traverse=False, **kwargs)
  File "/miniconda3/envs/xarray/lib/python3.7/site-packages/dask/base.py", line 399, in compute
    return repack([f(r, *a) for r, (f, a) in zip(results, postcomputes)])
  File "/miniconda3/envs/xarray/lib/python3.7/site-packages/dask/base.py", line 399, in <listcomp>
    return repack([f(r, *a) for r, (f, a) in zip(results, postcomputes)])
  File "/miniconda3/envs/xarray/lib/python3.7/site-packages/dask/array/core.py", line 828, in finalize
    return concatenate3(results)
  File "/miniconda3/envs/xarray/lib/python3.7/site-packages/dask/array/core.py", line 3621, in concatenate3
    result = np.empty(shape=shape, dtype=dtype(deepfirst(arrays)))
MemoryError
```

#### Output of ``xr.show_versions()``
<details>
# Paste the output here xr.show_versions() here
INSTALLED VERSIONS
------------------
commit: None
python: 3.7.3 (default, Mar 27 2019, 22:11:17) 
[GCC 7.3.0]
python-bits: 64
OS: Linux
OS-release: 4.15.0-51-generic
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8
libhdf5: 1.10.4
libnetcdf: 4.6.2

xarray: 0.12.2
pandas: 0.24.2
numpy: 1.16.4
scipy: 1.3.0
netCDF4: 1.5.1.2
pydap: None
h5netcdf: 0.7.3
h5py: 2.9.0
Nio: None
zarr: 2.3.2
cftime: 1.0.3.4
nc_time_axis: None
PseudonetCDF: None
rasterio: None
cfgrib: None
iris: None
bottleneck: 1.2.1
dask: 1.2.2
distributed: 1.28.1
matplotlib: 3.1.0
cartopy: None
seaborn: None
numbagg: None
setuptools: 41.0.0
pip: 19.1.1
conda: 4.7.5
pytest: None
IPython: 7.5.0
sphinx: None
</details>

### LINKED ISSUE #3277: xarray, chunking and rolling operation adds chunking along new dimension (previously worked)
I was testing the latest version of xarray (0.12.3) from the conda-forge channel and this broke some code I had.  Under the defaults installation not using conda-forge (xarray=0.12.1),  the following code works correctly with desired output:

#### Test code

```python
import pandas as pd
import xarray as xr
import numpy as np

s_date = '1990-01-01'
e_date = '2019-05-01'
days = pd.date_range(start=s_date, end=e_date, freq='B', name='day')
items = pd.Index([str(i) for i in range(300)], name = 'item')
dat = xr.DataArray(np.random.rand(len(days), len(items)), coords=[days, items])
dat_chunk = dat.chunk({'item': 20})
dat_mean = dat_chunk.rolling(day=10).mean()

print(dat_chunk)
print(' ')
print(dat_mean)

dat_std_avg = dat_mean.rolling(day=250).std()

print(' ')
print(dat_std_avg)
```

#### Output (correct) with xarray=0.12.1 - note the chunksizes
```
<xarray.DataArray (day: 7653, item: 300)>
dask.array<shape=(7653, 300), dtype=float64, chunksize=(7653, 20)>
Coordinates:
  * day      (day) datetime64[ns] 1990-01-01 1990-01-02 ... 2019-05-01
  * item     (item) object '0' '1' '2' '3' '4' ... '295' '296' '297' '298' '299'
 
<xarray.DataArray '_trim-8c9287bf114d61cb3ad74780465cd19f' (day: 7653, item: 300)>
dask.array<shape=(7653, 300), dtype=float64, chunksize=(7653, 20)>
Coordinates:
  * day      (day) datetime64[ns] 1990-01-01 1990-01-02 ... 2019-05-01
  * item     (item) object '0' '1' '2' '3' '4' ... '295' '296' '297' '298' '299'
 
<xarray.DataArray '_trim-2ee90b6c2f29f71a7798a204a4ad3305' (day: 7653, item: 300)>
dask.array<shape=(7653, 300), dtype=float64, chunksize=(7653, 20)>
Coordinates:
  * day      (day) datetime64[ns] 1990-01-01 1990-01-02 ... 2019-05-01
  * item     (item) object '0' '1' '2' '3' '4' ... '295' '296' '297' '298' '299'
```

#### Output (now failing) with xarray=0.12.3 (note the chunksizes)
```
<xarray.DataArray (day: 7653, item: 300)>
dask.array<shape=(7653, 300), dtype=float64, chunksize=(7653, 20)>
Coordinates:
  * day      (day) datetime64[ns] 1990-01-01 1990-01-02 ... 2019-05-01
  * item     (item) object '0' '1' '2' '3' '4' ... '295' '296' '297' '298' '299'
 
<xarray.DataArray (day: 7653, item: 300)>
dask.array<shape=(7653, 300), dtype=float64, chunksize=(5, 20)>
Coordinates:
  * day      (day) datetime64[ns] 1990-01-01 1990-01-02 ... 2019-05-01
  * item     (item) object '0' '1' '2' '3' '4' ... '295' '296' '297' '298' '299'

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
...

ValueError: For window size 250, every chunk should be larger than 125, but the smallest chunk size is 5. Rechunk your array
with a larger chunk size or a chunk size that
more evenly divides the shape of your array.
```

#### Problem Description
Using dask + rolling + xarray=0.12.3 appears to add undesirable chunking in a new dimension which was not the case previously using xarray=0.12.1  This additional chunking made the the queuing of a further rolling operation fail with a ValueError.  This (at the very least) makes queuing dask based delayed operations difficult when multiple rolling operations are used.

#### Output of ``xr.show_versions()`` for the not working version
<details>
INSTALLED VERSIONS
------------------
commit: None
python: 3.7.3 | packaged by conda-forge | (default, Jul  1 2019, 22:01:29) [MSC v.1900 64 bit (AMD64)]
python-bits: 64
OS: Windows
OS-release: 10
machine: AMD64
processor: AMD64 Family 23 Model 113 Stepping 0, AuthenticAMD
byteorder: little
LC_ALL: None
LANG: None
LOCALE: None.None
libhdf5: 1.10.4
libnetcdf: 4.6.1

xarray: 0.12.3
pandas: 0.25.1
numpy: 1.16.4
scipy: 1.3.1
netCDF4: 1.4.2
pydap: None
h5netcdf: None
h5py: 2.9.0
Nio: None
zarr: None
cftime: 1.0.3.4
nc_time_axis: None
PseudoNetCDF: None
rasterio: None
cfgrib: None
iris: None
bottleneck: 1.2.1
dask: 2.3.0
distributed: 2.3.2
matplotlib: 3.1.1
cartopy: None
seaborn: 0.9.0
numbagg: None
setuptools: 41.0.1
pip: 19.2.2
conda: 4.7.11
pytest: None
IPython: 7.8.0
sphinx: None
</details>

Apologies if this issue is reported, I was unable to find a case that appeared equivalent.

### PR DESCRIPTION
**Pull request by dcherian**
Should merge after https://github.com/dask/dask/pull/7234 is merged

- [x] Closes #3277, closes #2531, closes #2532, closes #2514
- [x] Tests added
- [x] Passes `pre-commit run --all-files`
- [x] User visible changes (including notable bug fixes) are documented in `whats-new.rst`

### CODE REVIEW

File: xarray/tests/test_duck_array_ops.py [hunk 1]
```diff
@@ -531,32 +531,6 @@ def test_isnull_with_dask():
     assert_equal(da.isnull().load(), da.load().isnull())
 
 
-@pytest.mark.skipif(not has_dask, reason="This is for dask.")
-@pytest.mark.parametrize("axis", [0, -1])
-@pytest.mark.parametrize("window", [3, 8, 11])
-@pytest.mark.parametrize("center", [True, False])
-def test_dask_rolling(axis, window, center):
```
Comment (dcherian) [inline xarray/tests/test_duck_array_ops.py:538]: moved to `TestVariableWithDask`

File: xarray/core/rolling.py [hunk 2]
```diff
@@ -502,13 +501,10 @@ def _bottleneck_reduce(self, func, keep_attrs, **kwargs):
 
         if is_duck_dask_array(padded.data):
             raise AssertionError("should not be reachable")
-            values = dask_rolling_wrapper(
```
Comment (dcherian) [inline xarray/core/rolling.py:505]: Should be undone. This is the bottleneck code path.

File: xarray/tests/test_variable.py [hunk 3]
```diff
@@ -873,26 +874,26 @@ def test_pad_constant_values(self, xr_arg, np_arg):
         )
         assert_array_equal(actual, expected)
 
-    def test_rolling_window(self):
+    @pytest.mark.parametrize("d, w", (("x", 3), ("y", 5)))
+    def test_rolling_window(self, d, w):
         # Just a working test. See test_nputils for the algorithm validation
         v = self.cls(["x", "y", "z"], np.arange(40 * 30 * 2).reshape(40, 30, 2))
-        for (d, w) in [("x", 3), ("y", 5)]:
```
Comment (dcherian) [inline xarray/tests/test_variable.py:879]: uses `parametrize` now

File: xarray/tests/test_nputils.py [hunk 4]
```diff
@@ -29,38 +29,3 @@ def test_vindex():
     vindex[[0, 1], [0, 1], :] = vindex[[0, 1], [0, 1], :]
     vindex[[0, 1], :, [0, 1]] = vindex[[0, 1], :, [0, 1]]
     vindex[:, [0, 1], [0, 1]] = vindex[:, [0, 1], [0, 1]]
-
-
```
Comment (dcherian) [inline xarray/tests/test_nputils.py:33]: moved to test_variable

**Comment by dcherian** [comment]
`sliding_window_view` dispatches with array_function so we could support other duck array libraries if they implemented it. cc @keewis

File: doc/duckarrays.rst [hunk 5]
```diff
@@ -44,8 +44,6 @@ the code will still cast to ``numpy`` arrays:
     addition to not supporting duck arrays in dimension coordinates
   * :py:meth:`Dataset.rolling_exp` and :py:meth:`DataArray.rolling_exp` (uses
     ``numbagg``)
-  * :py:meth:`Dataset.rolling` and :py:meth:`DataArray.rolling` (uses internal functions
-    of ``numpy``)
```
Comment (keewis) [inline doc/duckarrays.rst:48]: could we keep this and state that this requires `numpy=1.20`?

File: xarray/core/npcompat.py [hunk 6]
```diff
@@ -96,3 +97,125 @@ def __array_function__(self, *args, **kwargs):
...
+            required by the size of the sliding window.
+            That is, ``view.shape = x_shape_trimmed + window_shape``, where
+            ``x_shape_trimmed`` is ``x.shape`` with every entry reduced by one less
+            than the corresponding window size.
+
+        See Also
+        --------
+        lib.stride_tricks.as_strided: A lower-level and less safe routine for
+            creating arbitrary views from custom shape and strides.
+        broadcast_to: broadcast an array to a given shape.
```
Comment (keewis) [inline xarray/core/npcompat.py:159]:
```suggestion
        numpy.lib.stride_tricks.as_strided : A lower-level and less safe routine for
            creating arbitrary views from custom shape and strides.
        numpy.broadcast_to : broadcast an array to a given shape.
```

File: xarray/core/variable.py [hunk 7]
```diff
@@ -2078,26 +2078,39 @@ def rolling_window(
...
+            for arg in [window, window_dim, center]:
+                assert utils.is_scalar(arg)
             dim = [dim]
             window = [window]
             window_dim = [window_dim]
             center = [center]
+        else:
+            assert len(dim) == len(window)
+            assert len(dim) == len(window_dim)
+            assert len(dim) == len(center)
```
Comment (keewis) [inline xarray/core/variable.py:2096]: are these checks important? If they are, we might want to raise explicitly (`assert` is skipped with python's `-O` option)

File: xarray/core/variable.py [hunk 8]
```diff
@@ -2078,26 +2078,39 @@ def rolling_window(
...
+
+        pads = {}
+        for d, win, cent in zip(dim, window, center):
+            if cent:
+                start = int(win / 2)  # 10 -> 5,  9 -> 4
```
Comment (keewis) [inline xarray/core/variable.py:2101]:
```suggestion
                start = win // 2  # 10 -> 5,  9 -> 4
```

**Review by keewis — COMMENTED** [review]
I didn't do a thorough review, but I have a few comments.

> `sliding_window_view` dispatches with array_function

yes, this is really great.

File: xarray/core/npcompat.py [hunk 9]
```diff
@@ -96,3 +97,98 @@ def __array_function__(self, *args, **kwargs):
...
 IS_NEP18_ACTIVE = _is_nep18_active()
+
+
+if LooseVersion(np.__version__) >= "1.20.0":
+    sliding_window_view = np.lib.stride_tricks.sliding_window_view
+else:
+    from numpy.core.numeric import normalize_axis_tuple
+    from numpy.lib.stride_tricks import as_strided
+
+    def sliding_window_view(
```
Comment (mathause) [inline xarray/core/npcompat.py:108]:
```suggestion
    # copied from numpy.lib.stride_tricks
    def sliding_window_view(
```

File: xarray/core/dask_array_compat.py [hunk 10]
```diff
@@ -93,3 +93,125 @@ def nanmedian(a, axis=None, keepdims=False):
         )
 
         return result
+
+
+if LooseVersion(dask_version) > LooseVersion("2.30.0"):
+    ensure_minimum_chunksize = da.overlap.ensure_minimum_chunksize
+else:
+
+    def ensure_minimum_chunksize(size, chunks):
```
Comment (mathause) [inline xarray/core/dask_array_compat.py:102]:
```suggestion
    # copied from dask
    def ensure_minimum_chunksize(size, chunks):
```

File: xarray/core/dask_array_compat.py [hunk 11]
```diff
@@ -93,3 +93,125 @@ def nanmedian(a, axis=None, keepdims=False):
...
+                f"The overlapping depth {size} is larger than your "
+                f"array {sum(chunks)}."
+            )
+
+        return tuple(output)
+
+
+def sliding_window_view(x, window_shape, axis=None):
+    from dask.array.overlap import map_overlap
+    from numpy.core.numeric import normalize_axis_tuple
```
Comment (mathause) [inline xarray/core/dask_array_compat.py:156]:
```suggestion
    from numpy.core.numeric import normalize_axis_tuple  # type: ignore
```

**Review by mathause — COMMENTED** [review]
Looks good. I am still interested to eventually fix #2007. This PR may also make that easier as it now uses `Variable.pad`.

**Comment by dcherian** [comment]
> I am still interested to eventually fix #2007. This PR may also make that easier as it now uses Variable.pad.

That's the plan. This PR reduces the number of places we pad from 4 to 3 :P. The two bottleneck code paths still do their own padding.

File: xarray/core/npcompat.py [hunk 12]
```diff
@@ -96,3 +97,99 @@ def __array_function__(self, *args, **kwargs):
 
 
 IS_NEP18_ACTIVE = _is_nep18_active()
+
+
+if LooseVersion(np.__version__) >= "1.20.0":
+    sliding_window_view = np.lib.stride_tricks.sliding_window_view
+else:
+    from numpy.core.numeric import normalize_axis_tuple
```
Comment (mathause) [inline xarray/core/npcompat.py:105]:
```suggestion
    from numpy.core.numeric import normalize_axis_tuple  # type: ignore
```
there was a second one...

**Comment by dcherian** [comment]
The dask PR was merged so this should be ready to go.

**Comment by mathause** [comment]
It looked pretty good when I reviewed it, so :+1: 

Just quickly skimming the code it did not become entirely clear to me why this had to be after dask/dask#7234? You don't import `sliding_window_view` from dask. Also you re-define `sliding_window_view` in `dask_array_compat.py` - why don't you need this in dask proper. (Not complaining just wondering. Also feel free to not answer the question if you don't have time.)

**Comment by dcherian** [comment]
> You don't import sliding_window_view from dask.

haha. fixed!

I was waiting on that PR to see if the reviews would catch any bugs.

**Comment by andersy005** [comment]
@dcherian, should we test this in upstream-dev CI workflow (via the `[test-upstream]` keyword)?

**Comment by dcherian** [comment]
thanks @andersy005 !

**Pull request merged**